packages/web/src/utils/menus/utils.ts

```diff
@@ -0,0 +1,187 @@
+    }
+  // Enter || Spacebar
+  } else if (key == 13 || key == 32) {
+    if (activeElement.nodeName == "GCDS-MENU-GROUP") {
+      event.preventDefault();
+      if (activeElement.hasAttribute("open")) {
+        await closeMenuGroup(activeElement as HTMLGcdsMenuGroupElement, menu);
+      } else {
+        await openMenuGroup(activeElement as HTMLGcdsMenuGroupElement, menu);
+      }
+    }
+  }
+};
+  
+  
+/**
+* Open menu group
+* @param {Element} groupTrigger
+* @param {Element} menu
+*/
+async function openMenuGroup(groupTrigger, menu) {
+  await groupTrigger.toggleMenu();
+
+  if (groupTrigger.children[0].nodeName == "GCDS-MENU-GROUP") {
+    setTimeout(() => {
+      (groupTrigger.children[0] as HTMLGcdsMenuGroupElement).focusTrigger();
+    }, 10);
+  } else if (groupTrigger.children[0].nodeName == "GCDS-MENU-LINK") {
+    setTimeout(() => {
+      (groupTrigger.children[0] as HTMLGcdsMenuLinkElement).focusLink();
+    }, 10);
+  }
+
+  if (menu.nodeName == "GCDS-SIDEBAR-MENU") {
+    menu.updateMenuItemQueue(menu);
+  } else {
+    menu.updateMenuItemQueue(groupTrigger, true);
+  }
+}
+  
+/**
+* Close menu group
+* @param {Element} groupTrigger
+* @param {Element} menu
+*/
+async function closeMenuGroup(groupTrigger, menu) {
+  await groupTrigger.toggleMenu();
+  groupTrigger.focusTrigger();
+
+  menu.updateMenuItemQueue(menu);
+}
+  
+/**
+* Return array of child elements of passed element
+* @param {Element} el
+* @return {any[]} indexedItems
+*/
+export async function getMenuItems(el) {
+  let indexedItems: Array<any> = Array.from(el.children);
+
+  indexedItems.forEach(async (item) => {
+    if (item.nodeName == "GCDS-MENU-GROUP" && (item as HTMLGcdsMenuGroupElement).open) {
+      let groupChildren = await getMenuItems(item);
+      indexedItems.splice(indexedItems.indexOf(item) + 1, 0, ...groupChildren);
+    }
+  });
+
+  return indexedItems;
+}
+
+  /**
+* Create mobile menu element in passed menu
+* @param {HTMLElement} menu
+*/
+export async function configureMobileMenu(menu: HTMLElement) {
+    console.log(menu)
+  let children = Array.from(menu.children);
+
+  let menuGroup = document.createElement("gcds-menu-group") as HTMLGcdsMenuGroupElement;
+  menuGroup.heading = "Menu";
+  menuGroup.classList.add('gcds-mobile-menu');
+
+  for (let element of children) {
+    if (!element.hasAttribute("slot")) {
+      menuGroup.appendChild(element);
+    }
+  }
+
+  menu.appendChild(menuGroup);
+}
+
+/**
+* Unpack menu items from mobile menu in passed menu
+* @param {HTMLElement} menu
+*/
+export async function unpackMobileMenu(menu: HTMLElement) {
+  let mobileMenuChildren = [];
+  let mobileMenu: Element;
+
+  for (let item of Array.from(menu.children)) {
+    if (!item.hasAttribute("slot")) {
+      mobileMenuChildren = Array.from(item.children);
+      mobileMenu = item;
+    }
+  }
+
+  for (let element of mobileMenuChildren) {
+    menu.appendChild(element);
+  }
+
+  mobileMenu.remove();
+}
```

Comment:
Is it possible to reduce the repetition of code in this file? Potentially create functions for `focusLink`, etc.

It might also make sense to use a switch statement instead of multiple if conditions for handling the key events. It would make the code a bit easier to read.